pipreqs/pipreqs.py

```diff
@@ -111,9 +111,9 @@ def get_all_imports(
         candidates += [os.path.splitext(fn)[0] for fn in files]
         for file_name in files:
             file_name = os.path.join(root, file_name)
-            with open(file_name, "r", encoding=encoding) as f:
-                contents = f.read()
             try:
+                with open(file_name, "r", encoding=encoding) as f:
+                    contents = f.read()
                 tree = ast.parse(contents)
                 for node in ast.walk(tree):
                     if isinstance(node, ast.Import):
```

Comment:
I think there is still a bug here: `traceback.print_exc(exc)` (on line 126 just below) doesn't take the exception as argument, but the limit. So it should be `traceback.print_exc()` without an argument

Reply:
![image](https://github.com/user-attachments/assets/4d260d2d-900b-4c04-bf3b-a116b483c65a)
There are api changes in version 3.5 and 3.10